packages/opc-nav/src/opc-nav.ts

```diff
@@ -114,3 +72,108 @@ export class OpcNav extends LitElement {
     </header>`;
   }
 }
+
+@customElement('opc-nav-search')
+export class OpcNavSearch extends LitElement {
+  static get styles() {
+    return [style];
+  }
+
+  @property({ type: String, reflect: true })
+  value = '';
+
+  @state()
+  private _isSearchOpen: boolean = false;
+
+  @query('input')
+  private input!: HTMLInputElement;
+
+  private _handleSearchOpen() {
+    this._isSearchOpen = true;
+    this.input.focus();
+  }
+
+  private _handleSearchClose() {
+    if (!this.input.value) {
+      this.value = '';
+      this._isSearchOpen = false;
+    }
+  }
+
+  private _handleInputChange(e: Event) {
+    this.dispatchEvent(
+      new CustomEvent('opc-nav-search:change', {
+        detail: {
+          value: this.input.value,
+        },
+      })
+    );
+  }
+
+  private _handleSearchSubmit(e: Event) {
+    e.preventDefault();
+    this.dispatchEvent(
+      new CustomEvent('opc-nav-search:submit', {
+        detail: {
+          value: this.input.value,
+        },
+        bubbles: true,
+        composed: true,
+      })
+    );
+  }
+
+  render() {
+    const isSearchExpanded = Boolean(this.value) || this._isSearchOpen;
+
+    return html`
+      <div class="opc-nav-search">
+        <form
+          class="opc-nav-search__form"
+          @submit="${this._handleSearchSubmit}"
+          ?active=${isSearchExpanded}
+          @blur="${this._handleSearchClose}"
+        >
+          ${isSearchExpanded
+            ? html` <button class="opc-nav-search__btn" type="submit">
+                <img
+                  src="${searchIcon}"
+                  class="opc-nav__icon"
+                  alt="user"
+                  width="20px"
+                  height="20px"
+                />
+              </button>`
+            : html` <button
+                class="opc-nav-search__btn"
+                type="button"
+                @click="${this._handleSearchOpen}"
+              >
+                <img
+                  src="${searchIcon}"
+                  class="opc-nav__icon"
+                  alt="user"
+                  width="20px"
+                  height="20px"
+                />
+              </button>`}
+          <input
+            class="opc-nav-search__input ${isSearchExpanded
+              ? 'opc-nav-search__input-active'
+              : ''}"
+            type="search"
+            name="query"
+            autofocus
+            autocomplete="off"
+            aria-label="Search"
+            placeholder="Search"
+            required
+            .value=${live(this.value)}
```

Comment:
@akhilmhdh should it be without dot.
```suggestion
            value=${live(this.value)}
```

Reply:
https://lit.dev/docs/templates/directives/#live 
I implemented this with help of live directive documentation. They are adding . with template attribute even if string in example. I think its due to using directive